azure-iot-device/tests/provisioning/pipeline/test_pipeline_stages_provisioning.py

```diff
@@ -639,11 +639,11 @@ def stage(self, mocker, cls_type, init_kwargs):
 
 
 @pytest.mark.describe("RegistrationStage - .run_op() -- retried again with RegisterOperation")
-@pytest.mark.parametrize(
-    "request_payload",
-    [pytest.param(" ", id="empty payload"), pytest.param(fake_payload, id="some payload")],
-)
 class TestRegistrationStageWithRetryOfRegisterOperation(RetryStageConfig):
+    @pytest.fixture(params=[" ", fake_payload])
```

Comment:
>fixture [](start = 12, length = 7)

This change looks fine, but I'm curious about what inspired it and why you removed the ids.  At the surface, this looks like a style change, but I know that pytest can be tricky sometimes, especially with parametrization.

Reply:
Yes, please add the ids back for correct formatting.

---
In reply to: [392234226](https://github.com/Azure/azure-iot-sdk-python/pull/490#discussion_r392234226) [](ancestors = 392234226)

Reply:
pytest had changed something i am pretty sure for which the tests started failing. Mostly the reason is my prior parameters were tied to the class and even if the fixture using the params was not getting used in certain tests , it was asking me to declare it in those tests as well..this also led to the problem where the parent class needed the params/fixture even if there was no usage of those. now the params are tied to other fixtures , and not to class..

---
In reply to: [392234226](https://github.com/Azure/azure-iot-sdk-python/pull/490#discussion_r392234226) [](ancestors = 392234226)